apps/examples/err_report_demo/err_report_main.c

```diff
@@ -0,0 +1,329 @@
+	int nbytes_sent = 0;
+	int i;
+	int sock_cnt = 0;
+	int sock_fds[CONFIG_NSOCKET_DESCRIPTORS];
+
+	res = wifi_manager_init(&wifi_callbacks);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		printf(" wifi_manager_init fail\n");
+		goto done;
+	}
+	/* Create dummy sockets, the idea here is to
+	 * prevent dhcp from successfully opening a socket,
+	 * and then capture the resulting error.
+	 */
+	for (i = 0; i < CONFIG_NSOCKET_DESCRIPTORS; i++) {
+		sock_fds[i] = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+		if (sock_fds[i] < 0) {
+			/* We are done here, time to exit */
+			break;
+		} else {
+			sock_cnt++;
+		}
+	}
+	res = wifi_manager_connect_ap(&apconfig);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		printf(" AP connect failed\n");
+		goto done;
+	}
+	/* Wait for DHCP connection */
+	ERR_REPORT_TEST_WAIT;
+	/* Wait for Disconnected event, we must sleep here,
+	 * because WiFi Manager prevents callbacks to application
+	 * when disconnecting.
+	 */
+	sleep(10);
+
+	/* close dummy sockets created above */
+	for (i = 0; i < sock_cnt; i++) {
+		close(sock_fds[i]);
+	}
+
+	res = wifi_manager_connect_ap(&apconfig);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		printf(" AP connect failed\n");
+		goto done;
+	}
+	/* Wait for DHCP connection */
+	ERR_REPORT_TEST_WAIT;
+done:
+	/* Send error report for the bad AP connection */
+	nbytes_read = error_report_data_read(readbuf);
+	if (nbytes_read > 0) {
+		nbytes_sent = error_report_send(endpoint, readbuf, nbytes_read);
+	}
+	wifi_manager_disconnect_ap();
+	ERR_REPORT_TEST_WAIT;
+	wifi_manager_deinit();
+	ERR_REPORT_TC_END_CHECK(nbytes_sent > 0);
+}
+
+static void error_report_queue_underflow(void)
+{
+	ERR_REPORT_TC_START;
+
+	int nbytes_read;
+	char readbuf[1024];
+	nbytes_read = error_report_data_read(readbuf);
+
+	ERR_REPORT_TC_END_CHECK(nbytes_read == 0);
+}
+
+static void error_report_test(const char *endpoint)
+{
+	ERR_REPORT_LOG_START;
+	/* Verify unsuccessful data read */
+	error_report_queue_underflow();
+
+	/* Verify single error case in WiFi Manager */
+	error_report_single(endpoint);
+
+	/* Verify multiple errors across WiFi Manager */
+	error_report_multiple(endpoint);
+
+	ERR_REPORT_PRINT_STATS;
+	ERR_REPORT_LOG_END;
+}
+
+void error_report_process(int argc, char *argv[])
+{
+	if (argc < 3) {
+		printf("%s", ERR_REPORT_USAGE);
+		return;
+	}
+	if (strcmp(argv[2], "send") == 0) {
+		if (argc < 4) {
+			error_report_records(NULL);
+		} else {
+			error_report_records(argv[3]);
+		}
+	} else if (strcmp(argv[2], "test") == 0) {
+		if (argc < 4) {
+			error_report_test(NULL);
+		} else {
+			error_report_test(argv[3]);
+		}
+	} else {
+		printf("%s", ERR_REPORT_USAGE);
+	}
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int error_report_main(int argc, char *argv[])
+#endif
+{
+	printf("Network Error Report test!!\n");
+	sem_init(&g_err_sem, 0, 0);
+	g_err_nscenarios = 0;
+	g_err_nsuccess = 0;
+
+	task_create("Network Error Reporting", 100, 1024 * 10, (main_t) error_report_process, argv);
```

Comment:
Who gives argv? Is this working?

Reply:
argv comes from the TASH console. It has been verified.